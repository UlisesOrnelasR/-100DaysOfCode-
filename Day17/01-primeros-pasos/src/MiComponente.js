import React from "react";

const MiComponente = () => {

    let usuario = {
        nombre: 'pedro',
        apellidos: 'ornelas',
        web: 'www.pedro.es'
    }

    return (
        <>
            <h2>holaaaa</h2>
            <h3>Datos del usuario:</h3>
            <ul>
                <li>
                    Nombre:  <strong>{ usuario.nombre }</strong>
                </li>
                <li>
                    web: { usuario.web }
                </li>
            </ul>

        <p>Este es mi primer componente</p>
        </>
    )
}

export default MiComponente