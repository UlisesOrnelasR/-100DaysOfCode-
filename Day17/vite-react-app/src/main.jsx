import ReactDOM from 'react-dom';
import './index.css';

const app4 = <h1>Desarrollo util es lo maximo</h1>;
const container = document.getElementById('root');

ReactDOM.render(app4, container);
