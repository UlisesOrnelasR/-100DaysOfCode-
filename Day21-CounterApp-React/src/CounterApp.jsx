import { useState } from "react";
import PropTypes from "prop-types";

export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);
  const handleAdd = () => {
    //{console.log(event)};
    setCounter(counter + 1);
    //setCounter((c) => c + 1);
  };
  const handleRemove = () => {
    setCounter(counter - 1);
  };
  const handleReset = () => {
    setCounter(value);
  };
  return (
    <>
      <h1>CounterApp</h1>
      <h2> {counter} </h2>

      <button onClick={handleAdd}>AÃ±ade +1</button>
      <button onClick={handleRemove}>Quita -1</button>
      <button onClick={handleReset}>Reset</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
