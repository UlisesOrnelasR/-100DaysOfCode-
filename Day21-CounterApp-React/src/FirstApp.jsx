import PropTypes from "prop-types";

export const FirstApp = ({ title, subTitle, name }) => {
  // console.log(props);

  return (
    <>
      <h2>{title}</h2>
      <p>{subTitle}</p>
      <p>{name}</p>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
  name: PropTypes.string,
};

FirstApp.defaultProps = {
  title: "No hay titutlo",
  subTitle: "no hay subtitulo",
  name: "Ulises Ornelas",
};
