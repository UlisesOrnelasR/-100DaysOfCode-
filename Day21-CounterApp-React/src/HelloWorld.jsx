// export function HelloWorld() {
//   //document.createElement...
//   return <h1>Hola mundooo!!!</h1>;
// }

export const HelloWorld = () => {
  return <h2>HelloWorld</h2>;
};
