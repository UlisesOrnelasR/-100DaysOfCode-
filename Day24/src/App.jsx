export function App() {
  return (
    <>
      <h1>Practica para pruebas</h1>
      <h2>Insertando pruebas a componentes con mi blog</h2>
    </>
  );
}

export default App;
