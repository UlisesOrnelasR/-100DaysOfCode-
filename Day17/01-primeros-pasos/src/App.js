import logo from './logo.svg';
import './App.css';
import './MiComponente'
import MiComponente from './MiComponente';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Hola mundoooo
        </p>

        <MiComponente />
        
      </header>
    </div>
  );
}

export default App;
