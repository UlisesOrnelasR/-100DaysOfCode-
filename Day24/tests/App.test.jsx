import { App } from "./../src/App";
import { render, screen } from "@testing-library/react";

describe("Pruebas a <App />", () => {
  test("Deberia hacer snapshot con el componente", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
  test("Should haber una etiqueta h1", () => {
    render(<App />);
    expect(screen.getAllByText("Practica para pruebas")).toBeTruthy();
  });
});
