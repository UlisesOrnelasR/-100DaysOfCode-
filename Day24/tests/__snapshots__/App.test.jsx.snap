// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a <App /> Deberia hacer snapshot con el componente 1`] = `
<div>
  <h1>
    Practica para pruebas
  </h1>
  <h2>
    Insertando pruebas a componentes con mi blog
  </h2>
</div>
`;
